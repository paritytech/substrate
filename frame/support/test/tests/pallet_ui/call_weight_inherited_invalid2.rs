// Weight is an ident instead of a type.

use frame_support::pallet_prelude::*;
use frame_system::pallet_prelude::*;

pub trait WeightInfo {
	fn foo() -> Weight;
}

#[frame_support::pallet]
mod parentheses {
	use super::*;
	
	#[pallet::config]
	pub trait Config: frame_system::Config {
		type WeightInfo: crate::WeightInfo;
	}

	#[pallet::pallet]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::call(weight(prefix))]
	impl<T: Config> Pallet<T> {
		#[pallet::call_index(0)]
		pub fn foo(_: OriginFor<T>) -> DispatchResult {
			Ok(())
		}
	}
}

#[frame_support::pallet]
mod assign {
	use super::*;
	
	#[pallet::config]
	pub trait Config: frame_system::Config {
		type WeightInfo: crate::WeightInfo;
	}

	#[pallet::pallet]
	pub struct Pallet<T>(core::marker::PhantomData<T>);

	#[pallet::call(weight = prefix)]
	impl<T: Config> Pallet<T> {
		#[pallet::call_index(0)]
		pub fn foo(_: OriginFor<T>) -> DispatchResult {
			Ok(())
		}
	}
}

fn main() {
}
