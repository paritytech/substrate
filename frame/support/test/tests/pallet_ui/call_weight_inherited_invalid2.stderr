error[E0412]: cannot find type `prefix` in this scope
  --> tests/pallet_ui/call_weight_inherited_invalid2.rs:22:24
   |
22 |     #[pallet::call(weight(prefix))]
   |                           ^^^^^^ not found in this scope

error[E0412]: cannot find type `prefix` in this scope
  --> tests/pallet_ui/call_weight_inherited_invalid2.rs:43:26
   |
43 |     #[pallet::call(weight = prefix)]
   |                             ^^^^^^ not found in this scope
